Source/GSStoryboardTransform.m

```diff
@@ -0,0 +1,828 @@
+@end
+
+@interface NSStoryboardSegue (__StoryboardPrivate__)
+// Private to this class...
+- (void) _setKind: (NSString *)k;
+- (void) _setRelationship: (NSString *)r;
+- (NSString *) _kind;
+- (NSString *) _relationship;
+@end
+
+// this needs to be set on segues
+@implementation NSStoryboardSegue (__StoryboardPrivate__)
+- (void) _setKind: (NSString *)k
+{
+  ASSIGN(_kind, k);
+}
+
+- (void) _setRelationship: (NSString *)r
+{
+  ASSIGN(_relationship, r);
+}
+
+- (NSString *) _kind
+{
+  return _kind;
+}
+
+- (NSString *) _relationship
+{
+  return _relationship;
+}
+@end
+
+@implementation NSStoryboardSeguePerformAction
+- (id) target
+{
+  return _target;
+}
+
+- (void) setTarget: (id)target
+{
+  ASSIGN(_target, target);
+}
+
+- (SEL) action
+{
+  return _action;
+}
+
+- (void) setAction: (SEL)action
+{
+  _action = action;
+}
+
+- (NSString *) selector
+{
+  return NSStringFromSelector(_action);
+}
+
+- (void) setSelector: (NSString *)s
+{
+  _action = NSSelectorFromString(s);
+}
+
+- (id) sender
+{
+  return _sender;
+}
+
+- (void) setSender: (id)sender
+{
+  ASSIGN(_sender, sender);
+}
+
+- (NSString *) identifier
+{
+  return _identifier;
+}
+
+- (void) setIdentifier: (NSString *)identifier
+{
+  ASSIGN(_identifier, identifier);
+}
+
+- (NSString *) kind
+{
+  return _kind;
+}
+
+- (void) setKind: (NSString *)kind
+{
+  ASSIGN(_kind, kind);
+}
+
+- (NSStoryboard *) storyboard
+{
+  return _storyboard;
+}
+
+- (void) setStoryboard: (NSStoryboard *)storyboard
+{
+  ASSIGN(_storyboard, storyboard);
+}
+
+- (NSStoryboardSegue *) storyboardSegue
+{
+  return _storyboardSegue;
+}
+
+- (void) setStoryboardSegue: (NSStoryboardSegue *)ss
+{
+  ASSIGN(_storyboardSegue, ss);
+}
+
+- (void) dealloc
+{
+  RELEASE(_storyboard);
+  RELEASE(_kind);
+  RELEASE(_identifier);
+  RELEASE(_sender);
+  RELEASE(_storyboardSegue);
+  [super dealloc];
+}
+
+- (IBAction) doAction: (id)sender
+{
+  BOOL should = YES;
+
+  // If the instance we are testing is a controller, then the value of should is set by this method....
+  // if it is not, as it is possible to initiate a segue from an NSMenuItem, then we don't, but should
+  // remains set to YES so that the logic to replace the destination controller is still called.
+  if ([_sender respondsToSelector: @selector(shouldPerformSegueWithIdentifier:sender:)])
+    {
+      should = [_sender shouldPerformSegueWithIdentifier: _identifier
+                                                  sender: _sender];
+    }
+
+  if (should)
+    {
+      id destCon = [_storyboardSegue destinationController];
+      if ([destCon isKindOfClass: [NSString class]])
+        {
+          // resolve the destination controller
+          destCon = [_storyboard instantiateControllerWithIdentifier: destCon];
+          [_storyboardSegue _setDestinationController: destCon];  // replace with actual controller...
+        }
+      [_storyboardSegue _setSourceController: _sender]; 
+      
+      if (_sender != nil &&
```

Comment:
The call to prepareForSegue:sender: is missing here.

Reply:
Not true, currently it is called from within performSegueWithIdentifier:sender:. I am rather not sure whether this is the expected way to implement this